src/modules/SessionReplay.ts

```diff
@@ -75,3 +94,38 @@ export const setUserStepsEnabled = (isEnabled: boolean) => {
 export const getSessionReplayLink = async (): Promise<string> => {
   return NativeSessionReplay.getSessionReplayLink();
 };
+
+/**
+ * Set a callback for weather this session should sync
+ *
+ * @param handler
+
+ * @example
+ * ```ts
+ * SessionReplay.setSyncCallback((payload)=>{
+ *    if(payload.device == "Xiaomi M2007J3SY" &&
+ *         payload.os == "OS Level 33" &&
+ *         payload.appVersion == "3.1.4 (4)" ||
+ *         payload.sessionDurationInSeconds > 20 )
+ *    {return true}
+ * });
+ * ```
+ */
+export const setSyncCallback = async (
+  handler: (payload: sessionData) => boolean,
+): Promise<void> => {
+  emitter.addListener(NativeEvents.SESSION_REPLAY_ON_SYNC_CALLBACK_INVOCATION, (payload) => {
+    const result = handler(payload);
+    const shouldSync = Boolean(result);
+
+    if (typeof result !== 'boolean') {
+      console.warn(
+        `IBG-RN: The callback passed to SessionReplay.setSyncCallback was expected to return a boolean but returned "${result}". The value has been cast to boolean, proceeding with ${shouldSync}.`,
+      );
+    }
+
+    NativeSessionReplay.evaluateSync(Boolean(handler(payload)));
```

Comment:
We've already called the function and got the result above, no need to recall it here.

```suggestion
    NativeSessionReplay.evaluateSync(shouldSync);
```